pkg/lib/oauthrelyingparty/apple/provider.go

```diff
@@ -199,6 +215,41 @@ func (p Apple) GetUserProfile(ctx context.Context, deps oauthrelyingparty.Depend
 		return
 	}
 
+	// According to the documentation, the ID token NEVER include the name of the end-user.
+	// https://developer.apple.com/documentation/sign_in_with_apple/authenticating-users-with-sign-in-with-apple#Retrieve-the-users-information-from-Apple-ID-servers
+	//
+	// Instead, the name of the end-user is only available in the response when the end-user authorizes THE FIRST TIME.
+	// https://developer.apple.com/documentation/sign_in_with_apple/incorporating-sign-in-with-apple-into-other-platforms#Handle-the-response
+	//
+	// In THE FIRST TIME authorization, the form post request includes a form field named "user".
+	// This form field is a JSON document, with the following shape.
+	//
+	// { "name": { "firstName": "string", "lastName": "string" }, "email": "string" }
+	//
+	// Since "user" only appears THE FIRST TIME, you MUST revoke the authorization if
```

Comment:
@louischan-oursky If user only exist in the first time, later if the user sign in again, as the claims of the identity will be updated, the name will be disappeared?

Reply:
In fact, I seems didn't find any code that populate a "name" into the authgear standard user attribute. It seems what we populate is only `email`, `phone_number` and `username`.

Reply:
> If user only exist in the first time, later if the user sign in again, as the claims of the identity will be updated, the name will be disappeared?

Yes, I overlooked this. Our OAuth abstraction is now stateless, it does not have access to the existing profile. Maybe we can only add a special case in identity service to "merge" Apple user profile and standard attributes.

Reply:
> In fact, I seems didn't find any code that populate a "name" into the authgear standard user attribute. It seems what we populate is only email, phone_number and username.

See pkg/lib/feature/stdattrs/service.go:44 `PopulateStandardAttributes`

Reply:
Added a commit to address this problem.